packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -151,6 +153,7 @@ abstract class SerializationManager {
   /// Deserialize the provided json [data] by using the className stored in the [data].
   dynamic deserializeByClassName(Map<String, dynamic> data) {
     var className = data['className'];
+    if (className == 'null') return null;
```

Comment:
Chore: Change the validation so that we evaluate the type in the following order:
1. Primitive type
2. Project type
3. Module type.

After your first change we evaluate it in the following order:
1. Primitive type.
2. Module type.
3. Project type.

Reply:
this is not possible for "deserializeByClassName", only for "getClassNameForObject", because "deserializeByClassName" throws and error at the end of the method:
![Screenshot 2024-07-09 at 1 35 44 PM](https://github.com/serverpod/serverpod/assets/78809795/0cd3f765-f42d-4dc0-bd93-d837720361ad)

Reply:
This is fine in the deserialization 👍 It would just be an optimization to check the primitives first but won't be breaking in any way.